### PR TITLE
Updated AFNetworking to 3.x branch

### PR DESCRIPTION
**Pull request by miraclebg**
#### Ticket
**Required for Vimeans only**
[TICKET_NUMBER](https://vimean.atlassian.net/browse/TICKET_NUMBER)

#### Ticket Summary
This is a proposed update to switch from AFNetworking 2.x to 3.x.

#### Implementation Summary
Changes are only to swift files. I am a objective-c developer mostly but even though i don't think there are any significant AFNetworking changes - probably only swift related ones.

Feel free to blame me for any stupid mistakes :)

#### How to Test
Run the include demo and enjoy!

### CODE REVIEW

File: Podfile [hunk 1]
```diff
@@ -4,7 +4,7 @@ xcodeproj 'Example/VimeoUpload-iOS-OldUpload/VimeoUpload-iOS-OldUpload.xcodeproj
 xcodeproj 'Example/VimeoUpload-OSX/VimeoUpload-OSX.xcodeproj'
 
 def shared_pods
-    pod 'AFNetworking', '2.6.3'
+    pod 'AFNetworking', '~> 3.0'
```
Comment (alfiehanssen) [inline Podfile]: @miraclebg we should remove the `~>` so that we're setting one particular version number and aren't subject to the version changing automatically out from under us

Comment (alfiehanssen) [inline Podfile]: It looks like that would be 3.0.4

File: VimeoUpload.podspec [hunk 2]
```diff
@@ -21,7 +21,9 @@ Pod::Spec.new do |s|
   s.osx.deployment_target = '10.9'
 
   s.source       = { :git => "https://github.com/vimeo/VimeoUpload.git", :tag => s.version.to_s }
-  s.source_files  = "VimeoUpload", "VimeoUpload/**/*.{swift}"
+  s.source_files  = "*.{h}", "VimeoUpload", "VimeoUpload/**/*.{swift}", "VimeoUpload/**/*.{h}"
```
Comment (alfiehanssen) [inline VimeoUpload.podspec]: I don't think the changes to the podspec are necessary, we're actually not going to be able to make the library available as a pod until the Vimeo iOS app drops iOS7 support (sometime this summer probably).

File: VimeoUpload/Extensions/AFURLSessionManager+Extensions.swift [hunk 3]
```diff
@@ -42,6 +42,6 @@ extension AFURLSessionManager
     @available(iOS 8.0, *)
     func taskForIdentifier(identifier: Int) -> NSURLSessionTask?
     {
-        return self.tasks.filter{ $0.taskIdentifier == identifier }.first as? NSURLSessionTask
+        return self.tasks.filter{ $0.taskIdentifier == identifier }.first as NSURLSessionTask!
```
Comment (alfiehanssen) [inline VimeoUpload/Extensions/AFURLSessionManager+Extensions.swift]: We should revert this change, so that we don't crash if for some reason the cast fails...

File: VimeoUpload/Upload/Extensions/AFURLSessionManager+Upload.swift [hunk 4]
```diff
@@ -30,6 +30,6 @@ extension AFURLSessionManager
 {
     func uploadTaskForIdentifier(identifier: Int) -> NSURLSessionUploadTask?
     {
-        return self.uploadTasks.filter{ $0.taskIdentifier == identifier }.first as? NSURLSessionUploadTask
+        return self.uploadTasks.filter{ $0.taskIdentifier == identifier }.first as NSURLSessionUploadTask!
```
Comment (alfiehanssen) [inline VimeoUpload/Upload/Extensions/AFURLSessionManager+Upload.swift]: Same here, we should revert this change so we simply return nil in the event of a cast failure

File: VimeoUpload/Upload/Networking/Old Upload/VimeoSessionManager+OldUpload.swift [hunk 5]
```diff
@@ -117,7 +117,7 @@ extension VimeoSessionManager
     {
         let request = try (self.requestSerializer as! VimeoRequestSerializer).uploadVideoRequestWithSource(source, destination: destination)
         
-        let task = self.uploadTaskWithRequest(request, fromFile: source, progress: progress, completionHandler: { [weak self] (response, responseObject, error) -> Void in
+        let task = self.uploadTaskWithRequest(request, fromFile: source, progress: {(progress) -> Void in }, completionHandler: { [weak self] (response, responseObject, error) -> Void in
```
Comment (alfiehanssen) [inline VimeoUpload/Upload/Networking/Old Upload/VimeoSessionManager+OldUpload.swift]: Same here too, can you describe why this change is necessary? Seems like we should keep this logic as it was

**Comment by alfiehanssen** [comment]
@miraclebg Just reviewed the PR and left some comments, let me know what you think, seems totally possible to merge this AFNetworking 3.0.4 change into the v0.9 branch

File: VimeoUpload.podspec (hunk 2 above)
Comment (miraclebg) [inline VimeoUpload.podspec]: Please ignore all changes i've made to the Podfile. This was some late testing / tryouts to try to install the project through cocoapods (by setting the github repo param) but we never quite got it to work due to the VIMNetworking deps :) I don't know really why this appears here - i created the PR before doing these changes (i think).

File: Podfile (hunk 1 above)
Comment (miraclebg) [inline Podfile]: That's right - this was because i was trying to install through cocoapods and basically experiented a bit.

File: VimeoUpload/Extensions/AFURLSessionManager+Extensions.swift (hunk 3 above)
Comment (miraclebg) [inline VimeoUpload/Extensions/AFURLSessionManager+Extensions.swift]: screen shot 2016-04-06 at 7 14 41

I am not sure if you read above but i am in no way using swift on a daily basis - so i might just have done something stupid. But logic drived me to do what you see here :) Feel free to update it!

File: VimeoUpload/Upload/Extensions/AFURLSessionManager+Upload.swift (hunk 4 above)
Comment (miraclebg) [inline VimeoUpload/Upload/Extensions/AFURLSessionManager+Upload.swift]: Same issue as above here.

File: VimeoUpload/Upload/Networking/Old Upload/VimeoSessionManager+OldUpload.swift (hunk 5 above)
Comment (miraclebg) [inline VimeoUpload/Upload/Networking/Old Upload/VimeoSessionManager+OldUpload.swift]: As i understand in AF2 this is a return variable:

- (NSURLSessionUploadTask *)uploadTaskWithRequest:(NSURLRequest *)request
                                         fromFile:(NSURL *)fileURL
                                         progress:(NSProgress * __nullable __autoreleasing * __nullable)progress
                                completionHandler:(nullable void (^)(NSURLResponse *response, id __nullable responseObject, NSError  * __nullable error))completionHandler;

In AF3 - it is now a block var:

- (NSURLSessionUploadTask *)uploadTaskWithRequest:(NSURLRequest *)request
                                         fromFile:(NSURL *)fileURL
                                         progress:(nullable void (^)(NSProgress *uploadProgress)) uploadProgressBlock
                                completionHandler:(nullable void (^)(NSURLResponse *response, id _Nullable responseObject, NSError  * _Nullable error))completionHandler;

**Comment by miraclebg** [comment]
Hey @alfiehanssen - thanks for the review - just commented back.

Please once more :) do note i am and old school dev - i still love objective c, non-arc and non-xib development :) Someday i might give swift a chance - so apologizing for any stupid things you see above!

**Comment by alfiehanssen** [comment]
@miraclebg 

The upcoming v1.0 release will include AFNetworking v3.0, 

In light of that I'm going to close this PR, 

Other changes in v1.0 will include removal of iOS7 support, and full inline documentation, 

Thanks for bearing with us as we extricate iOS7 from our infrastructure,

**Comment by miraclebg** [comment]
Thank you for the brief!
